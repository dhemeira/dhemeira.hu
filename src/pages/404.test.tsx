import React from 'react';
import { cleanup, render, waitFor, screen } from '@testing-library/react';
import NotFoundPage from './404';

jest.mock('@gatsbyjs/reach-router', () => ({
  ...jest.requireActual('@gatsbyjs/reach-router'),
  useLocation: jest.fn().mockImplementation(() => ({
    pathname: '/mock-path',
  })),
}));

describe('Not Found Page', () => {
  afterEach(() => {
    cleanup();
  });

  test('renders correctly', async () => {
    const { container } = render(<NotFoundPage />);

    await waitFor(() =>
      expect(screen.queryByText(/The following page doesn't exist: /i)).toBeInTheDocument()
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
