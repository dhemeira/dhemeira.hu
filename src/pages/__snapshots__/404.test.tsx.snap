// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Page renders correctly 1`] = `
<div
  class="min-h-screen flex flex-col bg-light-background dark:bg-dark-background bg-top-gradient"
>
  <a
    class="fixed left-0 top-24 z-[100] p-5 pl-8 rounded-r-full bg-light-secondary/30 text-light-text text-center dark:bg-dark-secondary/30 dark:text-dark-text opacity-0 -translate-x-72 transition-all focus:opacity-100 focus:translate-x-0 focus:outline-none focus:ring dark:focus:ring-dark-accent focus:ring-light-accent"
    href="#main"
  >
    Skip to main
    <br />
    content
  </a>
  <header
    class="fixed top-0 w-full bg-black/75 z-50 backdrop-blur-xl text-dark-text border-b-white/20 border-b"
  >
    <div
      class="flex justify-between px-[5%] md:px-[10%] py-2"
    >
      <a
        class="font-medium px-2 text-base/8"
        href="/"
      >
        dhemeira
      </a>
      <nav
        class="flex gap-4 justify-end items-center"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/uni"
        >
          University
        </a>
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          aria-label="Navigation Menu"
          class="w-8 h-8 flex-col justify-center items-center gap-1 flex sm:hidden"
          data-headlessui-state=""
          id="headlessui-menu-button-:r2:"
          type="button"
        >
          <div
            class="w-8 h-1 bg-dark-text rounded transition-transform "
          />
          <div
            class="w-8 h-1 bg-dark-text rounded transition-transform "
          />
          <div
            class="w-8 h-1 bg-dark-text rounded transition-transform "
          />
        </button>
      </nav>
    </div>
    <div
      class="w-full translate-y-px"
    >
      <div
        class="h-0.5 bg-light-accent dark:bg-dark-accent"
        style="width: 0%;"
      />
    </div>
  </header>
  <main
    class="text-light-text dark:text-dark-text pt-24 md:pt-32 pb-12 flex-1"
    id="main"
  >
    <div
      class="container text-center flex flex-col justify-center items-center gap-8"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h1
            class="text-5xl sm:text-[6rem] font-bold tracking-tighter"
          >
            404
          </h1>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          The following page doesn't exist: /mock-path
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          Check out the home page to find what you are looking for
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <a
            class="text-base/9 font-semibold inline-block rounded-lg px-8 py-2 border-2 hover:bg-transparent transition hover:scale-105 hover:shadow-lg bg-light-primary dark:bg-dark-primary text-light-text dark:hover:text-dark-text border-light-primary dark:border-dark-primary  "
            data-headlessui-state=""
            href="/"
            type="button"
          >
            Back to home
          </a>
        </div>
      </div>
    </div>
  </main>
  <footer
    class="bg-black/75 w-full px-[5%] md:px-[10%] py-2 flex justify-between items-end text-dark-text border-t-white/20 border-t"
  >
    <span
      class="flex flex-col"
    >
      <a
        href="https://dhemeira.hu"
      >
        www.dhemeira.hu
      </a>
      <span>
        Â© 2023-
        2024
         | dhemeira
      </span>
    </span>
    <a
      aria-label="GitHub"
      class="p-1"
      href="https://github.com/dhemeira"
      rel="noreferrer"
      target="_blank"
      title="GitHub"
    >
      <svg
        class="w-5 h-5"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27s1.36.09 2 .27c1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.01 8.01 0 0 0 16 8c0-4.42-3.58-8-8-8"
        />
      </svg>
    </a>
  </footer>
</div>
`;
